### PR TITLE
Update gradle build of Azure CosmosDB connector

### PR DESCRIPTION
**Pull request by ayeshLK**
# Description
> $subject

Fixes: [#5785](https://github.com/ballerina-platform/ballerina-library/issues/5785)

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -12,57 +12,72 @@ license = ["Apache-2.0"]
...
+artifactId = "azure-cosmos"
+version = "4.52.0"
+path = "./lib/azure-cosmos-4.52.0.jar"
+
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-resolver-dns"
-version = "4.1.94.Final"
+version = "4.1.100.Final"
+path = "./lib/netty-resolver-dns-4.1.100.Final.jar"
```
Comment (NipunaRanasinghe) [inline ballerina/Ballerina.toml:34]: Is it mandatory to provide the path for maven dependencies? IINM it is not

File: build.gradle [hunk 2]
```diff
@@ -55,28 +65,56 @@ allprojects {
...
+        ballerinaStdLibs "io.ballerina.stdlib:task-ballerina:${stdlibTaskVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:cache-ballerina:${stdlibCacheVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:file-ballerina:${stdlibFileVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:mime-ballerina:${stdlibMimeVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:auth-ballerina:${stdlibAuthVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:jwt-ballerina:${stdlibJwtVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:oauth2-ballerina:${stdlibOAuth2Version}"
+        ballerinaStdLibs "io.ballerina.stdlib:http-ballerina:${stdlibHttpVersion}"
+        ballerinaStdLibs "io.ballerina.stdlib:observe-ballerina:${observeVersion}"
+        ballerinaStdLibs "io.ballerina:observe-ballerina:${observeInternalVersion}"
```
Comment (NipunaRanasinghe) [inline build.gradle:95]: Do we need explictly define all the stdlib deps? IINM we can depend only on the distribution
(@ThisaruGuruge correct me if I'm missing something 🙂)

File: .gitignore [hunk 3]
```diff
@@ -1,67 +1,45 @@
...
+ballerina/.settings/
+build-config/checkstyle/.classpath
+build-config/checkstyle/.project
+build-config/checkstyle/.settings/
+native/.classpath
+native/.project
+native/.settings/
+test-utils/.classpath
+test-utils/.project
+test-utils/.settings/
```
Comment (ThisaruGuruge) [inline .gitignore:45]: Why do we need all these locations?

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (ThisaruGuruge) [reply ballerina/Ballerina.toml:34]: If we don't provide the path, each Ballerina build will download the dependencies. This way, we handle the maven dependencies through the Gradle build, which is efficient.

File: build.gradle (hunk 2 above)
Comment (ThisaruGuruge) [reply build.gradle:95]: +1. For connectors, we don't need the intermediate Ballerina pack. (At least for now :D)

File: ballerina/build.gradle [hunk 4]
```diff
@@ -33,46 +51,114 @@ def stripBallerinaExtensionVersion(String extVersion) {
...
+    }
+}
+
+apply plugin: 'io.ballerina.plugin'
+
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    langVersion = ballerinaLangVersion
+    testCoverageParam = "--code-coverage --coverage-format=xml --includes=*"
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:64]: Do we need the `includes=*` here?

Comment (ThisaruGuruge) [inline ballerina/build.gradle:64]: For a connector build, we should pass the `buildOnDockerImage` parameter.

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (NipunaRanasinghe) [reply ballerina/Ballerina.toml:34]: didn't know that. it makes sense now :) thanks for the insight

File: .gitignore (hunk 3 above)
Comment (ayeshLK) [reply .gitignore]: Fixed with [2bfd8e2899585fa713ca440a10210084ab1282cf](https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/112/commits/2bfd8e2899585fa713ca440a10210084ab1282cf)

File: build.gradle (hunk 2 above)
Comment (ayeshLK) [reply build.gradle]: Fixed with [656be44575890fe06cc09c85ab80315abe3e392f](https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/112/commits/656be44575890fe06cc09c85ab80315abe3e392f)

File: ballerina/build.gradle (hunk 4 above)
Comment (ayeshLK) [reply ballerina/build.gradle]: Fixed with [acfafde7659710d8b170eca0f0509ba4a6cdc6fa](https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/112/commits/acfafde7659710d8b170eca0f0509ba4a6cdc6fa)

File: build.gradle [hunk 5]
```diff
@@ -17,14 +17,20 @@
...
 }
 
 description = 'Ballerina - Azure Cosmosdb'
 
+ext.ballerinaLangVersion = project.ballerinaLangVersion
+ext.azureCosmosVersion = project.azureCosmosVersion
+ext.nettyVersion = project.nettyVersion
+ext.reactorNettyVersion = project.reactorNettyVersion
+ext.micrometerVersion = project.micrometerVersion
+
```
Comment (ThisaruGuruge) [inline build.gradle:33]: We don't need these assignments.

File: gradle.properties [hunk 6]
```diff
@@ -1,14 +1,47 @@
...
+# Level 04
+stdlibAuthVersion=2.10.0
+stdlibJwtVersion=2.10.0
+stdlibOAuth2Version=2.10.0
+
+# Level 05
+stdlibHttpVersion=2.10.3
+
+observeVersion=1.2.0
+observeInternalVersion=1.2.0
```
Comment (ThisaruGuruge) [inline gradle.properties:47]: Why do we need these dependencies? They should be handled by the bal tool.

File: ballerina/build.gradle [hunk 7]
```diff
@@ -16,10 +16,28 @@
...
+            credentials {
+                username System.getenv("packageUser")
+                password System.getenv("packagePAT")
+            }
+        }
+    }
+    dependencies {
+        classpath "io.ballerina:plugin-gradle:${project.ballerinaGradlePluginVersion}"
+    }
+}
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:34]: As a best practice, let's do all the plugin management inside the settings.gradle file. Check the example repo (Gmail).

File: build.gradle (hunk 5 above)
Comment (ayeshLK) [reply build.gradle]: Fixed with [2280283f9bc1a8ab83cff0aedcaff14e6b45e0ca](https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/112/commits/2280283f9bc1a8ab83cff0aedcaff14e6b45e0ca)

File: ballerina/build.gradle (hunk 7 above)
Comment (ayeshLK) [reply ballerina/build.gradle]: Fixed with [2f2d71b9726944ad42af3ecdae198f45b1687811](https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/112/commits/2f2d71b9726944ad42af3ecdae198f45b1687811)

**Review by ThisaruGuruge — DISMISSED** [review]
LGTM

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**